front/src/modules/activities/tasks/hooks/useTasks.ts

```diff
@@ -87,7 +87,7 @@ export function useTasks(entity?: ActivityTargetableEntity) {
   });
 
   const tasksData =
-    activeTabId === 'done' || !entity ? completeTasksData : incompleteTaskData;
+    activeTabId === 'to-do' || entity ? incompleteTaskData : completeTasksData;
```

Comment:
We want to show incompleteTasks if the current tab is to-do OR if we are on an entity page, I re-arrange the condition to avoid confusions. I think this hook is still confusing though but let's mitigate the issue before doing too much changes